### ISSUE TITLE
Render disused:railway=rail like railway=disused

### ISSUE THREAD

**Issue by Penegal** [gravitystorm/openstreetmap-carto#2030]
Hello, there. Hope you're not growing tired of my suggestions :wink: 

Shouldn't we render `disused:railway=rail` like `railway=disused`? This new scheme seems to be expanding, but, as the default rendering does not render them, it doesn't support wide adoption by naive contributors, which may think "It does not display if I tag it that way, only if tagged the old way, so let's go with the old way". Besides, not rendering it means that disused railways converted to the new scheme are no longer rendered, which seems a regress to me: this feature is often an important landmark when present.

Regards.

**Comment by mboeringa** [comment]
I think you may actually be seriously **misinterpreting** the whole purpose of a **"disused:x=y"** versus a ***"x=disused"*** tag. As I understood from e.g. the *"Lifecycle prefix"* and *"Comparison of life cycle concepts"* webpages:

http://wiki.openstreetmap.org/wiki/Lifecycle_prefix
http://wiki.openstreetmap.org/wiki/Comparison_of_life_cycle_concepts

the whole purpose of using **"disused:x=y"** contrary to using a ***"x=disused"***, is exactly the result you are now getting: ***no*** rendering of features that users have explicitly set to **"disused:x=y"**, so as to avoid showing objects / functions that may have previously existed, but are no longer there. 

E.g., to better understand this, see the reference to the **disused:amenity=pub** example, where it would be ludicrous to show a former pub as a pub on the map if it has been converted to e.g. a normal living house or shop.

Based on this, I think the current rendering ***shouldn't*** be changed, but rather kept.

This also mains that the **disused:railway=rail** tag is *not* a replacement for the **railway=disused** tag, but rather that these tag should live next to each other, and be used appropriately based on the local mappers desire to show or hide certain disused railways for applications.

**Comment by math1985** [comment]
No, the *=disused tagging is a bad scheme and should not be used. A data consumer shouldn't need to filter out disused objects by checking the disused tag.

I think the suggestion by @Penegal makes sense.

**Comment by mboeringa** [comment]
> No, the *=disused tagging is a bad scheme and should not be used. A data consumer shouldn't need to filter out disused objects by checking the disused tag.

I wasn't responding to the **x=disused** in general tag, but to the well established and much used **railway=disused** tagging scheme.

I think this tag, in case of the railways, is actually used in the sense of the **amenity=pub** example in the lifecycle pages, and thus I think what I previously wrote about the **disused:railway=rail** versus **railway=disused** is relevant.

> I think the suggestion by @Penegal makes sense.

It only makes sense if you would ***explicitly desire*** to render previous / former functions of objects in openstreetmap-carto / any application, not if you think it is a replacement for **railway=disused**.

E.g. do you suggest to render stuff like **disused:amenity=pub** and possibly thousands of other **disused:x=y** objects as well in carto?

Of course, this may be a custom rendering decision made by the openstreetmap-carto team for a specific case. There is nothing wrong with that, any rendering of OpenStreetMap data is to a large extent interpretation, but don't make it light-hearted.

**Comment by planemad** [comment]
@mboeringa
>   no rendering of features that users have explicitly set to "disused:x=y", so as to avoid showing objects / functions that may have previously existed, but are no longer there

The [disused](http://wiki.openstreetmap.org/wiki/Key:disused:) features still exist on the ground and are valuable to show in a map for navigational purposes.

**Comment by mboeringa** [comment]
> The disused features still exist on the ground and are valuable to show in a map for navigational purposes.

I never suggested to **not** render **railway=disused**, which is the current and most used tag for this. In fact, I suggest quite the opposite. I only suggest to not render **disused:railway=rail**, as rendering it is a wrong interpretation of OpenStreetMap lifecycle tagging IMO.

If you read carefully, the fact that it "is valuable to show in a map" is exactly why I compared the **railway=disused** tag with **amenity=pub**. Contrary to what the **disused:x=y** tag (e.g. disused:railway=rail or disused:amenity=pub) was designed for, people ***actually want to see the*** **railway=disused** ***tag rendered***. In contrast with this, **disused:x=y** as per the http://wiki.openstreetmap.org/wiki/Comparison_of_life_cycle_concepts page, was actually designed to register a *previous* function of an object, that should not normally rendered by default (but could in a specialized renderer targeted at disused objects, which I don't think openstreetmap-carto is...).

In addition, since **railway=disused** uses the main key **railway=x** (which is also comparable with the amenity=pub example), and not the much more problematic and deprecated **disused=yes** tag, which uses a secondary key to signify the disused status and thus causes problems with rendering of only a main key, it is also less of problem to decide whether or not to include it in rendering. You can simply include or exclude the **railway=disused** class from rendering, contrary to having to examine a secondary key as with the disused=yes deprecating tagging.

**Comment by planemad** [comment]
@mboeringa ah, thanks for clarifying.

**Comment by matkoniecz** [comment]
> the whole purpose of using "disused:x=y" contrary to using a "x=disused", is exactly the result you are now getting: no rendering of features that users have explicitly set to "disused:x=y"

For start: this is tagging for renderer approach. disused:x=* makes sense to use if disused object is vastly different from one that is not disused. For example shop=florist is POI where one may buy flowers, it is open at some times etc. disused:shop=florist has none of this functions and it is (if at all) used for completely different purposes.

With railways situation is different also for additional reason - here railway=disused scheme is widely used ([73 176 times](http://taginfo.openstreetmap.org/tags/railway=disused)) unlike [5594 usages worldwide for disused:railway](http://taginfo.openstreetmap.org/keys/disused%3Arailway).

In that situation I propose to start from discussion on tagging mailing list and/or with railway mappers whatever using railway:disused instead/in addition to railway=disused is a desirable idea.

In that situation starting from rendering is a poor idea.

On the other hand I see that OpenRailwayMap [is already rendering this tag] (https://github.com/rurseekatze/OpenRailwayMap/search?utf8=%E2%9C%93&q=disused%3Arailway&type=Code) (I expected opposite), so it may be accepted more than I expected.

**Comment by dieterdreist** [comment]
sent from a phone

> Am 24.03.2016 um 15:01 schrieb Mateusz Konieczny <notifications@github.com>:
> 
> For start: this is tagging for renderer approach. disused:x=* makes sense to use if disused object is vastly different from one that is not disused. For example shop=florist is POI where one may buy flowers, it is open at some times etc. disused:shop=florist has none of this functions and it is (if at all) used for completely different purposes.


I understand the shop tag as tagging a function: it is not the physical structure (space) tagged like this, but it's the business, someone selling flowers. The tag disused:shop=florist is something I'd maybe use in case the florist has moved/closed but there are still traces (e.g. signage). If there's a different business now in that shop it feels like unnecessary clutter to add references to the history (i.e. what there was before)

**Comment by aceman444** [comment]
Yes, the problem here is that contrary to disused:shop (that now sells something else or nothing), for disused:railway=* the rails do still exist on the ground and can be rendered.

Also, we wiki for key:railway itself proposes disused:railway as alternative tagging.

**Comment by mboeringa** [comment]
> Yes, the problem here is that contrary to disused:shop (that now sells something else or nothing), for disused:railway=* **the rails do still exist on the ground and can be rendered.**

Which is 

- 1) A possible reason to re-tag them to their former **railway=disused** _if you desire to see it rendered._ using the current, IMO correct, render rules of Carto.

_Or(!)_

- 2)  To **_double_** tag them with both:

**railway=disused**
and
**disused:railway=x**

if you seek to get it rendered and at the same time also register the previous _type of the railway_.

But the rendering should only be based on the **railway=disused** tag as - as I tried to explain before - rendering based on **disused:railway=x** is a mis-interpretation of OpenStreetMap lifecycle tagging.

Using the **disused:railway=x** _alone_ as the basis for rendering is really not recommended (unless in some specialized renderer targeted at disused objects).

**disused:railway=x** is not a replacement of **railway=disused**, these two tags should co-exist.

**Comment by matkoniecz** [comment]
> if you desire to see it rendered

That is tagging for renderer.

**Comment by mboeringa** [comment]
> That is tagging for renderer - http://wiki.openstreetmap.org/wiki/Tagging_for_the_renderer
> 
> Whatever X will be rendered in Y should not influence tagging decisions.

No, you are still not getting my point.

This is not tagging for the renderer. It is putting back the tag that Carto supports in rendering for **railway=disused**, _that may have been replaced by disused:railway=x alone_, although people actually want to see it rendered.

The whole purpose of a **disused:x=y** tagging scheme is to hide the feature for current applications, and to register a previous function, see the OpenStreetMap lifecycle tagging page:

http://wiki.openstreetmap.org/wiki/Comparison_of_life_cycle_concepts

QUOTE:
> By using this key prefix it is ensured that current or legacy applications are not confused by objects which do not exist or are not fully functional and only software aware of this tagging concept will evaluate those. 

Carto IMO **justly** does not render **disused:railway=x**, and justly renders only **railway=disused**. 

Both tags should co-exist, and it is up to the local OpenStreetMap community to decide if they want to use one or the other tagging scheme (or double tag with both tags), and thus either decide to show disused railways on the map (which by far the majority of railway mapping enthusiasts seem to want), or to hide the railways but just have the features in the OSM database for specialized applications.

**Comment by matkoniecz** [comment]
> thus either decide to show disused railways on the Standard Map - which by far the majority of railway mapping enthusiasts seem to want - or to hide the railways

That is tagging for renderer - tagging should describe reality, not decide what should be rendered. At this stage that is not yet "Don't deliberately enter data incorrectly for the renderer", but what happens for group that decided to enforce

> hide the railways

once somebody adds a correct railway=disused tag? Remove correct tag because they want to change rendering? That would be pure tagging for renderer ("Don't deliberately enter data incorrectly for the renderer").

**Comment by mboeringa** [comment]
Please @matkoniecz, actually start reading and trying to understand all the posts I wrote in this thread instead of letting your emotions reign... 

I never suggested to "hide the railways" once

> somebody adds a correct railway=disused tag?

In fact, I suggest _no changes at all_ to current Carto rendering! (which is correct in my opinion).

I also certainly did not suggest to

> "deliberately enter data incorrectly for the renderer"

However, I **_do_** think entering **disused:railway=x** in the database, **_and then asking for it to be rendered by Carto_**, is incorrect, because it violates the OpenStreetMap lifecycle tagging proposal.

**Comment by dieterdreist** [comment]
sent from a phone

> Am 27.03.2016 um 15:10 schrieb mboeringa <notifications@github.com>:
> 
> However, I do think entering disused:railway=x in the database, and then asking for it to be rendered by Carto, is incorrect, because it violates the OpenStreetMap lifecycle tagging proposal.
> 

what are you referring to? Surely not this concept: http://wiki.openstreetmap.org/wiki/Lifecycle_prefix

**Comment by mboeringa** [comment]
> Surely not this concept: http://wiki.openstreetmap.org/wiki/Lifecycle_prefix

Yes. Unless you can point me to another page describing **disused:railway=x** and more generally **disused:x=y** lifecycle tagging. Yes, the page may be thin in some aspects, but it is better than nothing and does make sense (well, to me at least...)

Of course, there is the other page as well:

http://wiki.openstreetmap.org/wiki/Comparison_of_life_cycle_concepts

**Comment by dieterdreist** [comment]
sent from a phone

> Am 27.03.2016 um 17:28 schrieb mboeringa <notifications@github.com>:
> 
> Surely not this concept: http://wiki.openstreetmap.org/wiki/Lifecycle_prefix
> 
> Yes. Unless you can point me to another page describing disused:railway=x and more generally disused:x=y lifecycle tagging. Yes, the page may be thin in some aspects, but it is better than nothing and does make sense (well, to me at least...)
> 


then I might have misunderstood you: how is that page in contradiction to tagging disused:railway=rail?

**Comment by dieterdreist** [comment]
sent from a phone

> Am 27.03.2016 um 11:00 schrieb mboeringa <notifications@github.com>:
> 
> The whole purpose of a disused:x=y tagging scheme is to hide the feature for current applications, and to register a previous function, see the OpenStreetMap lifecycle tagging page:


I disagree, the purpose is not to hide those features generally, but to avoid confusion by showing them like an in-use feature when you don't check for (a theoretically infinite list of) additional modifiers that change the meaning of other tags. It is a safety measure to show these different features only if you want to and not by accident.

**Comment by mboeringa** [comment]
> then I might have misunderstood you: **how is that page in contradiction to tagging disused:railway=rail**?

It's not. 

I have tried to explain this: I am _not_ against tagging either **railway=disused** or **disused:railway=rail**. Both have their function, they should co-exist.

In my opinion though, only **railway=disused** should be rendered by a general map like Carto, and **disused:railway=rail** used either as a secondary tag to document and specify the type of the disused railway in the OSM database, or used in specialized renderers / styles that target **disused:x=y** tagging.

**Comment by mboeringa** [comment]
> It is a safety measure to show these different features only if you want to and not by accident.

That is exactly what I wanted to say, but wrote in different words

**Comment by Penegal** [comment]
@mboeringa: I understand your arguments, at least I think so, but I still don't understand why rendering `disused:railway=*` is a problem. I mean, I agree that the `disused:` —and related life cycle tags— tagging scheme is supposed to ease the masking of unnecessary, not yet or no longer used, amenities and features, but I don't understand why `railway=disused` should be added to a `disused:railway=*`: IMO, these tags mean the same thing, but the second gives more freedom to tag the railway characteristics, for example if it was a narrow gauge railway, while the first essentially only tells that the railway was disused. Used this way, the life cycle prefix tagging scheme makes sense, but I still does not understand why, in this specific case of railway life cycle, your approach is not tagging for the renderer, because I still don't understand why using two tags which essentially say the same thing: the railway is disused. On this point, I think @dieterdreist is right, but, again, I may have missed your point despite your numerous attempts to make it clear.

**Comment by mboeringa** [comment]
@Penegal 
Yes, unfortunately it seems difficult.

My main objections or points regarding rendering based on **disused:railway=x** are the following:

- If Carto starts rendering disused:x=y tags, where do you stop? There is potentially thousands of ordinarily used tags that could just as well be rendered from a **disused:x=y** tag. None of these represent current functions. Bottom line: what makes **disused:railway=x** so special that it would deserve this special treatment over something like **disused:amenity=pub** or any of the other thousands of disused:x=y tags? 

**Note:** for the sake of the argument I am trying to take a completely neutral position / stance here, which isn't easy as I love trains too... but hey, there may be big historic pub lovers too!

- **railway=disused** is currently rendered in Carto and is generally used in the sense of **amenity=pub**. That is, **_people actually expect to see it rendered_**. This is contrary to what the **disused:x=y** tagging lifecycle tagging scheme describes. The **disused:x=y** tagging scheme was proposed as a way to describe former functions of objects, while avoiding the pitfall of requiring each and every current renderer to filter out objects with functions that may no longer exist. It would be a real problem if the map showed thousands of former shops...

- As per the above, I really consider rendering based on **disused:x=y** a mis-interpretation of the OpenStreetMap lifecycle tagging _for any general purpose style_... 

- There is nothing wrong with specialized styles designed to show former functions using **disused:x=y** (like OpenRailwayMap seems to be doing according to some posts here)!

- I don't say, tag one and not the other, I say: tag **railway=disused** if your local railway enthusiasts community wants to see the disused railway tracks rendered in the Standard Map / Carto. If not, use **disused:railway=x**, _or_ use it to add extra detail / information about the type to a **railway=disused** (just like we use secondary keys in many cases in OSM to add extra detail or refine keys, e.g. **natural=water**, **water=x**).

- In essence, in my opinion and probably my most controversial statement here, and based on the current and past tagging practices of railway mappers,

**railway=disused**
**disused:railway=x**

is very much equivalent to

**natural=water**
**water=x**

- I also think **disused:railway=x** is **not** a replacement for **railway=disused** as most people seem to think, just like **water=x** is not a replacement for **natural=water**. These tags should co-exist. Thinking disused:x=y is a replacement is the basis of the problem in understanding my argument.

- I propose no change to current rendering, which is correct in my opinion.

Lastly: I don't say there couldn't be an exception for the railway, but personally, I don't see the advantage outweighs the disadvantage of the clash with OpenStreetMap lifecycle tagging. There is very little to gain switching the rendering from **railway=disused** to **disused:railway=x**, while there is clearly something to lose (compliance with lifecycle tagging, setting a precedence for many more requests for rendering of disused:x=y in Carto).

I just say: think well before making this decision.

(and this will be my last ramble about this subject here...)

**Comment by Penegal** [comment]
> I also think disused:railway=x is not a replacement for railway=disused as most people seem to think, just like water=x is not a replacement for natural=water. These tags should co-exist. Thinking disused:x=y is a replacement is the basis of the problem in understanding my argument.

OK, so this is where I went lost in your argumentation. Indeed, your point of view makes sense on this: `disused:railway=x` can be seen as a refinement of `railway=disused`; as `water=*` is a refinement of `natural=water`. I don't think it's the case, though, as I never read that the life cycle prefixed tags should be added to the current tags, but I understand your reasoning.

**Comment by dieterdreist** [comment]
sent from a phone

> Am 27.03.2016 um 23:56 schrieb mboeringa <notifications@github.com>:
> 
> railway=disused is currently rendered in Carto and is generally used in the sense of amenity=pub. That is, people actually expect to see it rendered.


I believe it hasn't yet been pointed out that  railway=disused is also not a nice tag from a semantic point of view, as the key railway alone does not say "rails" but "railway related", and besides rail has values like platform, level_crossing, signal, station, tram, funicular, monorail, miniature etc.
http://wiki.openstreetmap.org/wiki/Key:railway

it would be much better if it were railway=disused_rail (or track)

Also see common values for the disused key: http://taginfo.osm.org/keys/disused#values
E.g. disused=station is quite common (if you leave yes/no out), and while we might not care if a disused railway is actual rails or a tram, we will surely not want to display disused stations the same like a disused track.

**Comment by RicoZ-OSM** [comment]
following up on @dieterdreist : 

Exactly. As far as I understand the documentation the railway=disused is "legal" only for rail-values but not the many other values like station, stop etc. So for a disused railway=station I think it would be more correct to use disused:railway=station than railway=disused . Whether or how  such objects should be rendered at all is a different question.

**Comment by jojo4u** [comment]
From the data perspective *=disused is the same as disused:*=*.
As pointed out railway has it's semantic problems with "disused", but this is not enough to break up the rule above.

**Comment by Gazer75** [comment]
Any progress on this? Having rail that is not used for any traffic show same as a regular rail line is not a good thing. It might lead people to thing there is a scheduled service.

**Comment by nebulon42** [comment]
Depends on #2533. This PR is one of the changes that has quite some impact for OSM. Help there is much appreciated.

**Comment by mboeringa** [comment]
> From the data perspective =disused is the same as disused:=*.

I still don't agree, but I think I have given enough reasons for this in several posts here, and seem to stand alone on this... (especially see https://github.com/gravitystorm/openstreetmap-carto/issues/2030#issuecomment-202154226. For additional arguments, please read the whole thread )

**Comment by Gazer75** [comment]
I agree with you. I see no reason why it can't just be railway=disused. I believe this was rendered before?

With current rendering I am more inclined to just go with abandoned to avoid it being rendered as regular rail.

**Comment by boothym** [comment]
Unless something has changed dramatically, railway=disused is still rendered and should be used for disused railways ("A section of railway which is no longer used but where the track and infrastructure remain in place." according to the wiki). If there's no passengers, freight or other services then re-tag it as railway=disused.

**Comment by Gazer75** [comment]
It is rendered like regular rail, that's the problem. Maybe it always has been?

**Comment by nebulon42** [comment]
Two points: Please keep this issue to `disused:railway=rail` as this is the topic. I'm not aware that `railway=disused` is rendered like regular railway. See e.g. https://www.openstreetmap.org/way/32531887.

**Comment by mboeringa** [comment]
So, to re-cap:
- If you want to tag disused railways **_and want to see them displayed on the map_**, use **railway=disused** which is rendered on the Carto / Standard map.
- If you want to tag disused railways **_and DON'T want them to show up on the map_**, use **disused:railway=x** _ONLY_.

In case you used **railway=disused**, you CAN potentially additionally add disused:railway=x to add detail about the railway type, like water=x adds detail to natural=water, adding this tag _won't_ affect the rendering, as the current rendering is solely based on railway=disused.

In case you used **disused:railway=x** to **_hide_** the features according to OpenStreetMap lifecycle tagging, you SHOULDN'T add railway=disused, as it doesn't add extra info, and it will make the feature show up on the map

**Comment by SomeoneElseOSM** [comment]
@mboeringa - you seem to be suggesting the choice between "disused:x=y" and "x=disused" is "whether the mapper wants it rendered or not".  I'm not convinced by that idea, but this github issue surely isn't the best place for that discussion :)

Maybe write a diary entry about it?

**Comment by mboeringa** [comment]
> @mboeringa - you seem to be suggesting the choice between "disused:x=y" and "x=disused" is "whether the mapper wants it rendered or not".

No, this is [OpenStreetMap lifecycle tagging](http://wiki.openstreetmap.org/wiki/Comparison_of_life_cycle_concepts).

But yes, the consequence is that you need to make a choice. A choice that ultimately results in the feature being visible or not in current Carto / Standard map rendering.

And I do think this is utterly relevant for this discussion and issue thread. It hits the core of the (non-existing IMO) problem...

**Comment by SomeoneElseOSM** [comment]
To be clear, https://wiki.openstreetmap.org/wiki/Comparison_of_life_cycle_concepts is "an overview of various methods used to tag the life-cycle of features" - it's not a hard-and-fast series of rules.  I could give plenty of edge cases of whether something is still there or not (e.g. a canal that still has water in it but isn't used for transport - whether it's disused depends on what you think it is used for).  I don't think that this is the place for that discussion though.

**Comment by mboeringa** [comment]
I just don't get it: people want to see disused railways rendered, and then use a tagging scheme designed to hide disused objects from existing applications and rendering implementations, _while there is a tag that does render disused railways_ and that doesn't really clash with OpenStreetMap lifecycle tagging because it is used as amenity=pub: **railway=disused**.

Would you also find it logical to tag your favorite pub with the tag:

**this:is:my:favourite:amenity=pub**

and ask the Carto team to render it because:

**amenity=pub**

also renders?

**Comment by SomeoneElseOSM** [comment]
As I understand it the tagging argument for "disused:railway=rail" rather than "railway=disused" is so that you can have e.g. "disused:railway=narrow_gauge".  It's absolutely nothing to do with whether or not something renders - that's a map style decision on a feature by feature basis.

**Comment by mboeringa** [comment]
> As I understand it the tagging argument for "disused:railway=rail" rather than "railway=disused" is so that you can have e.g. "disused:railway=narrow_gauge".

You still **_can_** by refining the tagging with the additional tag. This is even recommended on the official [railway=disused](http://wiki.openstreetmap.org/wiki/Tag:railway%3Ddisused) Wiki page (and no: I have not written that page).

It is also what I recommended in https://github.com/gravitystorm/openstreetmap-carto/issues/2030#issuecomment-295188287 and https://github.com/gravitystorm/openstreetmap-carto/issues/2030#issuecomment-202154226.

If the whole issue just revolves around: we are not willing to tag any additional keys... than I am flogging dead horses.

**Comment by Gazer75** [comment]
Looks like the problem is adding disused:railway=rail will make it render as regular rail regardless of railway=disused.

Similar with abandoned:railway=rail/light_rail/tram. Add that to a way and it will render as regular rail, even if railway=abandoned is there.

Another and similar issue is the newer tagging of preserved railways. Instead of railway=preserved you use railway:preserved=yes + railway=rail/light_rail/tram.
Using just railway=preserved give no option to tell you what kind of railway.

Basically there is need to update the rendering of newer tagging system.

**Comment by mboeringa** [comment]
> Looks like the problem is **adding disused:railway=rail will make it render as regular rail regardless of railway=disused**.

This _**would**_ be  a problem, but looking at the features and examples I link below, I think this is not the case and again based on a misunderstanding. Look at these examples with the corresponding tagging (after you verified the actual tagging, remove the way information and orange highlight line by clicking on the "X" cross above the way information: this will reveal the actual rendered feature):

**railway=disused + disused:railway=rail**:

- https://www.openstreetmap.org/way/45396635
- https://www.openstreetmap.org/way/486825256
- https://www.openstreetmap.org/way/28962459

They are clearly shown as railway=disused with a broken line symbol. Not like regular railway.

The Overpass Turbo query to get these results: http://overpass-turbo.eu/s/osh

I also do not see any references in this thread to actual features showing this behavior.

> Similar with abandoned:railway=rail/light_rail/tram. Add that to a way and it will render as regular rail, even if railway=abandoned is there.

Do you have concrete example of this?, because these features don't show up at all, they are not rendered:

**railway=abandoned + abandoned:railway=rail**:

- https://www.openstreetmap.org/way/424531418
- https://www.openstreetmap.org/way/470030570
- https://www.openstreetmap.org/way/409948227

The Overpass Turbo query to get these results: http://overpass-turbo.eu/s/osi

**Comment by Gazer75** [comment]
Ok after checking again it appears someone thought it was smart to add railway=rail to the relation for some of these. This will mess up rendering.

**Comment by matkoniecz** [comment]
Note https://github.com/gravitystorm/openstreetmap-carto/blob/master/CARTOGRAPHY.md

> It's an important feedback mechanism for mappers to validate their edits and helps to prevent unfavorable fragmentation of tag use.

Is there evidence that `disused:railway=rail` is preferred over `railway=disused` and `railway=disused disused=rail`?

----

@matthijsmelissen 

> No, the *=disused tagging is a bad scheme and should not be used. A data consumer shouldn't need to filter out disused objects by checking the disused tag.

I agree that data consumers should not need to check `disused=*` tag - for example `shop=supermarket disused=yes` is a horrible tagging. Disused value also may be problematic - for example `shop=vacant` is a poor idea, as "render all shops" is now requiring more complicated queries.

But in case of railways I think that `railway=disused` is OK. The simplest "render all railway lines" is not broken here and to get any more detailed information here it is necessary anyway to check value of `railway` tag.

-----

Overall, I think that supporting `disused:railway=rail` would just encourage tag fragmentation and is not desirable.

**Comment by Hedaja** [comment]
I support the rendering of life cycle prefixes for certain tags. Railways are one of them, the other one would be buildings.
I never read anywhere in the wiki that disused:* is for if you don't want to render it and *=disused is for having it render. That also would go against one of the main principals of OSM....DON'T TAG FOR THE RENDERER!

*=disused/construction/... is just the legacy form of tagging which is also still wildly adapted.
disused:* and other prefixes are an extension and simplification of the legacy life cycle concept (as far as I understand)

And from a data creator perspective, I can tell you its the much nicer concept to work with. If I have roadworks or works on a railway in this case, I simply add the prefix and when it's done, I delete it. All subinformation about the highway/railway type just stays the same the whole time.
If I on the other hand work with railway=construction, information gets lost and I need to add another tag construction=* to add it again. Same in reverse. If one forgets to add the construction/disused tag, the previously available information gets lost (I know you can look into the history to recover it but this can go unnoticed for quite a while)

I don't understand why disused or abandoned things tagged with the prefix shouldn't be rendered. In railways =disused might still be more common but for example in buildings [abandoned:building=*](https://taginfo.openstreetmap.org/keys/abandoned%3Abuilding#overview) is much more common than [building=abandoned](https://taginfo.openstreetmap.org/tags/building=abandoned)

**Comment by jeisenbe** [comment]
I checked the current trends. Since mid 2018 to now, there have been 11000 `railway=disused` ways added, versus 5000 disused:railway=* features (all values - including ways). There are also over 6 times as many `railway=disused` features overall (from [Taghistory](https://taghistory.raifer.tech)):

![way-railway-disused-vs-disused-railway](https://user-images.githubusercontent.com/42757252/65373925-d9f32b80-dcbe-11e9-9081-45f2fb5767f9.png)

The length of ways added was about equal at 10,000 kilometers, but it looks like most of the disused:railway kilometers were added all at once, probably by one user or one import (from [ohsome.org](https://ohsome.org/apps/dashboard/)):

![disused-railway-length-12-months](https://user-images.githubusercontent.com/42757252/65373956-3e15ef80-dcbf-11e9-8d41-e6af35c7aa39.png)
(Note non-zero Y axis, unfortunately not adjustable on ohsome dashboard)

![railway-disused-12-months-ohsome](https://user-images.githubusercontent.com/42757252/65373989-b381c000-dcbf-11e9-9048-bd99824db960.png)

Over 10,000 ways are tagged with both disused:railway and railway=disused, so only about 5000 ways have only disused:railway. See https://overpass-turbo.eu/s/MtY

I will close this issue for now, because disused=railway is still being used by more mappers.

If this trend reverses, and disused:railway becomes more common than railway=disused, then we should reopen this issue.

**Comment by dieterdreist** [comment]
I believe osm-carto should render alternative tags equally when there is
agreement that the alternatives are equivalent, around for some time and
are used in significant numbers (e.g. here they are used significantly).

**Comment by jeisenbe** [comment]
@dieterdreist, this would conflict with 2 of the main purposes of this style:

- It's an important feedback mechanism for mappers to validate their edits and helps to prevent unfavorable fragmentation of tag use.
...
- It's an exemplar stylesheet for rendering OSM data.
https://github.com/gravitystorm/openstreetmap-carto/blob/master/CARTOGRAPHY.md

Usually this means it is best to only render one way of tagging a certain feature.

**Comment by Gazer75** [comment]
Keep in mind that railway=disused blocks the ability to tell what kind of railway(light_rail, tram...). That is not a problem with disused:railway=

**Comment by kocio-pl** [comment]
Sorry for a technical interruption, but how one does flag the comment as off-topic? (of course you can also tag this comment as such :smiley: ).

**Comment by pnorman** [comment]
"Hide" under the `...` menu, then select a reason.

**Comment by Gazer75** [comment]
Key is to encourage use of the lifecycle prefixes as those will not conflict with others.
https://wiki.openstreetmap.org/wiki/Lifecycle_prefix

Yes we should not tag for the renderer.
But like it or not, this project/style impacts greatly how people tag.

**Comment by mboeringa** [comment]
> Edit: This is very on topic. By not rendering disused:railway= as such you promote bad tagging practice.
> I see wiki says railway=disused + disused:railway=* but that is just a double tag that is pointless.

No, quite on the contrary, we encourage good tagging practices. 

It is actually starting to render any **disused:x=y** tags that would encourage bad tagging practices and open up essentially pandora's box, _because where do we stop_??? While **disused:railway=x** is maybe your personal pet project, other users might request theirs to be rendered. There could be thousands of them.

Would you personally like to see all **disused:amenity=fuel** on the map, sending you of main road into some tiny hamlet with a near empty tank only to find out the petrol station you saw rendered on the map has long been abandoned and the edifice converted to a living space or just derelict? **_Likely not..._**

These tags and lifecycle prefixes are to record a past state, WITHOUT affecting current renderings and rendering engines for OpenStreetMap data. By starting to render them, we would encourage a plethora of other requests for rendering of **disused:x=y** or even **abandoned/razed:x=y**.

The problem is that people simply do not recognize **railway=disused** for what it is: a clearly visible stretch of land with some former railway usage **that people see as a structural topographic element** preferentially needing to be displayed on any map having the pretenses of being a "topographic" style map 

As such, I would dearly have wished that this tag had never been called **railway=disused**, but rather something like **railway=trackbed**, but alas, that is history...

**Comment by Hedaja** [comment]
> > Edit: This is very on topic. By not rendering disused:railway= as such you promote bad tagging practice.
> > I see wiki says railway=disused + disused:railway=* but that is just a double tag that is pointless.
> 
> No, quite on the contrary, we encourage good tagging practices.
> 
> It is actually starting to render any **disused:x=y** tags that would encourage bad tagging practices and open up essentially pandora's box, _because where do we stop_??? While **disused:railway=x** is maybe your personal pet project, other users might request theirs to be rendered. There could be thousands of them.
> 
> Would you personally like to see all **disused:amenity=fuel** on the map, sending you of main road into some tiny hamlet with a near empty tank only to find out the petrol station you saw rendered on the map has long been abandoned and the edifice converted to a living space or just derelict? **_Likely not..._**
> 
> These tags and lifecycle prefixes are to record a past state, WITHOUT affecting current renderings and rendering engines for OpenStreetMap data. By starting to render them, we would encourage a plethora of other requests for rendering of **disused:x=y** or even **abandoned/razed:x=y**.
> 
> The problem is that people simply do not recognize **railway=disused** for what it is: a clearly visible stretch of land with some former railway usage **that people see as a structural topographic element** _preferentially needing to be displayed on any map having the pretenses of being a "topographic" style map_.
> 
> As such, I would dearly have wished that this tag had never been called **railway=disused**, but rather something like **railway=trackbed**, but alas, that is history...

That doesn't really make much sense to me. If you are against rendering disused things in general, why do we render railway=disused?
disused:railway=... is exactly the same and gets used for clearly visible disused railways too.

You don't have to render all disused: tags just because you do it for one. In some cases that might make sense but in others it would just not be helpful/confusing.

As Gazer just said above your post. We shouldn't tag for the render but people will still do. Especially when there are two legitimate tagging schemes for the same thing. You therefore will always have the chicken egg problem when an old tagging scheme is trying to be changed. The numbers would probably start to change if renderers/editors would start to support it.

I can't see what disadvantages the life cycle tagging would have compared to the older tagging scheme, but I see advantages and therefore don't see a reason not to support it.

**Comment by mboeringa** [comment]
> That doesn't really make much sense to me. If you are against rendering disused things in general, why do we render railway=disused?
> disused:railway=... **is exactly the same** and gets used for clearly visible disused railways too.

If you read carefully and had made an effort to follow my reasoning, you would have recognized that I am actually arguing that these are NOT the same, and that the general tagging practice for disused railways is _that people actually **want** to see them displayed_ (which is counter to the function of the lifestyle tagging scheme).

**railway=disused** (in my opinion better named railway=trackbed)

**_is not the same as_**:

**disused:railway=x**

IMO.

**Comment by boothym** [comment]
>The problem is that people simply do not recognize railway=disused for what it is: a clearly visible stretch of land with some former railway usage that people see as a structural topographic element preferentially needing to be displayed on any map having the pretenses of being a "topographic" style map.

> As such, I would dearly have wished that this tag had never been called railway=disused, but rather something like railway=trackbed, but alas, that is history...

While I agree with everyone else you've said, this bit is incorrect - you are getting confused with railway=abandoned. 😃 That is different to railway=disused which is for tagging a section of railway where the rails and infrastructure is still in place but they are not being used.

**Comment by mboeringa** [comment]
> And what you describe as a disused railway, is already a razed railway. Disused railways still have all their tracks in place and could be used again without to much effort. What you describe would be :razed:railway=... :)

> While I agree with everyone else you've said, this bit is incorrect - you are getting confused with railway=abandoned.

Where in my description, did you see me mentioning anything about the presence or absence of the actual railway tracks (which would determine if it is was possibly abandoned / razed)?

I specifically refrained from mentioning the status of tracks being present:

> "a clearly visible stretch of land with some former railway usage **that people see as a structural topographic element**"

This description still leaves the option open of tracks being present or not (and I intend it to be read as such ;-) ).

**Comment by Gazer75** [comment]
> I checked the current trends. Since mid 2018 to now, there have been 11000 `railway=disused` ways added, versus 5000 disused:railway=* ways. There are also over 6 times as many `railway=disused` features overall (from [Taghistory](https://taghistory.raifer.tech)):

This is proof that because disused:railway=* is not showing people will not use it, and instead opt for railway=disused because that is rendered.

**Comment by mboeringa** [comment]
> This is proof that because disused:railway=* is not showing people will not use it, and instead opt for railway=disused because that is rendered.

Which is _exactly_ my point: people WANT disused railway to render, **hence using lifestyle tagging and disused:railway=x is not appropriate**.

**Comment by Gazer75** [comment]
And why is that not appropriate? It is exactly what the lifecycle prefix is for.
It removes the need for multiple tags to describe an object.

**Comment by mboeringa** [comment]
> And why is that not appropriate? It is exactly what the lifecycle prefix is for.
> It removes the need for multiple tags to describe an object.

Sorry, I should rephrase to make this clear:

Which is exactly my point: people WANT disused railway to render, hence using lifestyle tagging and disused:railway=x **on its own** is not appropriate. It needs additional tagging like railway=disused to force rendering **_if desired._**

> AFAIK the openrailway people are loosing their hair over this thing as they need the railway= value to be a proper one to show what kind of rail to render, even if it is not in use.

I think we **all** will start loosing our hair when people start massively requesting rendering of hundreds of **possible disused/abandoned/razed:x=y** lifecycle tags

I also do not see what the problem is for the OpenRailwayMap people, if people tag **railway=disused & disused:railway=rail**, they CAN render the appropriate type. That people are lazy enough to refuse to add the appropriate double tag, is another issue altogether.

**Comment by Gazer75** [comment]
People want disused rail to render and because the default renderer is not showing disused:railway= they have to use railway=disused. No idea why you can't see this.

Having to use multiple tags to describe objects when it has a perfectly valid single tag is not a good thing IMO.
It kind of like people using the blanket access=no for a road with no through traffic for motor vehicles, and then add a long list of tags with access for different modes. They could simply have done motor_vehicle=destination and be done with it.
Why do they do this? Because the default renderer will render the access=no and access=destination differently, but not motor_vehicle=destination. This is kind of tagging for the renderer even though they are using documented tags.

**Comment by Hedaja** [comment]
I don't understand. Could you point me to a source where it is documented that railway=disused and disused:railway= are not the exact same thing?

The life cycle prefixes are as far as I know exactly a replacement for the existing tags.

For example is construction:highway the same as highway=construction and so on

**Comment by camelCaseNick** [comment]
The wiki says

> Use the disused: lifecycle prefix on tags that relate to features that are in a reasonable state of repair but which are currently unused.
– https://wiki.openstreetmap.org/wiki/Key:disused:

While disused pubs are of no importance for OpenStreetMap, disused railways, highways and buildings stand out in reality and hence are of importance to OpenStreetMap and henceforth openstreetmap-carto.

Because we would never map for the renderer, the idea that `disused:` might tell the renderer to not render something is bs.

This tagging scheme just provides the possibility of marking specific tags as describing something that was and not blocking the value part of the key value pair (a.k.a. tag).

TL;DR:
`disused:` does not mean, it shouldn't be rendered.
It means it still exists, but is unused and would need repair.

**Comment by jeisenbe** [comment]
I appreciate the interesting comments about how disused railways should be tagged, but this is not the best place to discuss how features should be tagged. The [tagging mailing list](https://lists.openstreetmap.org/listinfo/tagging) or [forum.openstreetmap.org](https://forum.openstreetmap.org) would be good places for discussion. If someone wishes to deprecate railway=disused and replace this tag with disused:railway=* the process is described at https://wiki.openstreetmap.org/wiki/Proposal_process

**Comment by dieterdreist** [comment]
sent from a phone

> On 24. Sep 2019, at 23:10, mboeringa <notifications@github.com> wrote:
> 
> This description still leaves the option open of tracks being present or not (and I intend it to be read as such ;-) ).


the term “disused” in the OpenStreetMap context implies that the tracks are still there and in a reasonable state. Abandoned also implies the tracks are still there , but in a degenerate state. If you speak explicitly about trackbed as determining characteristics, it seems to imply that the tracks aren’t there anymore.

**Comment by mboeringa** [comment]
> **They both have the exact same meaning**, but the latter was probably made before people thought about the different kinds of disused rail.

I repeat, I do not agree with this bold statement that they _"are the exact same thing"_. 

You also contradict yourself in the sense that on the one hand you  are saying that **railway=disused** and **disused:railway=x** are the exact same thing, but on the other hand your recognize that only the **disused:railway=x** tag provides the additional information about the **type** of the railway (_"to show what kind of rail to render"_).

This observation by yourself clearly demonstrates these tags are **not** _"the exact same thing",_ even in your own view.

> I don't understand. Could you point me to a source where it is documented that railway=disused and disused:railway= are not the exact same thing?

The lifecycle page here explicitely states one of the main purposes of the lifecycle prefixes:
https://wiki.openstreetmap.org/wiki/Comparison_of_life_cycle_concepts

Quote:
_"By using this key prefix it is ensured that current or legacy applications are not confused by objects which do not exist or are not fully functional and only software aware of this tagging concept will evaluate those."_

Although this sentence clearly does not explicitly state that lifecycle prefixed should not be rendered, it does raise the important question that only the openstreetmap-carto developers can answer:

**Does openstreetmap-carto need to become, or is it desirable for it to become, a _"software aware of this tagging concept"??_**

I think it will open up a potential can of worms with a plethora of similar discussions like this one in the repository about the desirability of rendering any lifecycle prefixed objects.

I personally would prefer to see these lifecycle objects rendered dynamically as an extra layer on top of a rendered map, similar to e.g. http://openpoimap.org/,  but that requires adjustments to the OpenStreetMap website.

> **While disused pubs are of no importance for OpenStreetMap**, disused railways, highways and buildings stand out in reality and hence are of importance to OpenStreetMap and henceforth openstreetmap-carto.

That is your personal opinion. There may well be someone interested in the history of a region who **is** interested in locations of disused historic pubs. In fact, I have heard about one project that exactly did this: try to extract locations of historic (disused / abandoned / razed) pubs from OpenStreetMap data for the publication of a book about a town's history.

**Comment by dieterdreist** [comment]
Am Mi., 25. Sept. 2019 um 01:22 Uhr schrieb Nick Name <
notifications@github.com>:

> While disused pubs are of no importance for OpenStreetMap, disused
> railways, highways and buildings stand out in reality and hence are of
> importance to OpenStreetMap and henceforth openstreetmap-carto.




a disused pub is not an amenity=pub anymore, but a disused building is
still a building=*, context and semantics matter. Physical "things" remain
regardless of being actually used or not, "abstract" things like businesses
and services aren't there anymore when "disused". To decide an individual
case it must be clear what the tags represent.

Taking the example of railway=rail, I just learnt from the wiki it means
"trains" ;-) (I'm fixing this now)

> Use railway <https://wiki.openstreetmap.org/wiki/Key:railway>=rail for
> full sized passenger or freight trains in the standard gauge for the
> country or state.
>

Or according to the short definition, it means rails:

> Rails of a standard gauge track
>

So logically and from the wiki, disused railway tracks could well be
railway=rail disused=yes
https://wiki.openstreetmap.org/wiki/Tag:railway%3Drail

Or it could be railway=disused disused=rail or disused:railway=rail
(synonymous), for example if the legal status also should be accounted for
(and they aren't legally active rails anymore).

**Comment by Gazer75** [comment]
> I appreciate the interesting comments about how disused railways should be tagged, but this is not the best place to discuss how features should be tagged.

This issue is not about that, but the fact that carto do not render disused:railway=* as railway=disused if the latter is missing.
And mboeringa seem to be incapable of seeing that lifecycle prefix is better than adding multiple tags to describe this. Thus (s)he is not willing to accept that a simple one tag like this should be rendered.

**Comment by imagico** [comment]
> And mboeringa seem to be incapable of seeing that lifecycle prefix is better than adding multiple tags to describe this.

As @jeisenbe already said: Please stop discussing tagging questions here.

There are exactly two ideas w.r.t. disused railways we could discuss here:

* ceasing to render railway=disused
* rendering disused:railway=* instead of railway=disused.

The latter is currently out of the question given the use numbers presented.  The former would of course be an option but i would suggest to open a new issue for that if you think it is a serious consideration.

As @jeisenbe explained the idea of rendering both tags as synonyms would not be compatible with the goals of this style.

**Comment by Gazer75** [comment]
> The latter is currently out of the question given the use numbers presented. The former would of course be an option but i would suggest to open a new issue for that if you think it is a serious consideration.

This is exactly why new tags are never adopted. People running this seem to be unable to accept new tags before they have very high usage, thus causing people to use older tagging to make the object render.

**Comment by Hedaja** [comment]
>As @jeisenbe explained the idea of rendering both tags as synonyms would not be compatible with the goals of this style.

I thought this style was meant for mappers espacially. So displaying commonly used tags and tags that the cummunity has agreed upon.
life cycle tags are agreed upon and documented as an alternative to the old tagging scheme and are already adopted. even if not as widely as the old scheme which is bot a surprise if it doesn't get rendered.

**Comment by imagico** [comment]
I completely understand that there are people who would like to see this style take a more active role in steering mappers to map the 'right' way.  But trying to initiate such a change in overall direction by lobbying for making individual changes against established and documented goals is not going to work.  Discussions of ideas for overall policy changes are welcome (in a separate issue of course) as long as they focus on generic arguments and are not just pushed as an instrument to facilitate specific individual changes.

**Comment by mboeringa** [comment]
> thus causing people to use older tagging to make the object render.



> life cycle tags are agreed upon and documented as an alternative to the old tagging scheme

It is a misconception to think of lifecycle tags as being a "new" way to tag things, it is an additional attribute of an object instead.

The ultimate consequence of such reasoning would be that we would need drop all existing tags and to introduce a whole new lifecycle prefix for **_all_** OpenStreetMap tags, e.g. something named like "current", to denote that objects are current and in active use according to tag:

**current:amenity=pub**
**current:power=line**
...
**current:railway=rail**
etc.

I hope you agree this is absolutely ludicrous.

**Comment by Gazer75** [comment]
Now you're being a silly nitpicker. Where is "current" mentioned?

Double tagging is a terrible idea and is the whole reason to use only prefix.
It is clear you're not a fan of these and that is fine.

**Comment by Hedaja** [comment]
I would have to dive deeper on who initiated the life cycle prefixes. But it's meant to unify tagging life cycles.
But that doesn't even really matter here.
What matters is that both schemes describe the same thing and can even be used combined (because there is not a clear way of adding detail to *=disused).
And the prefix is widely established
https://wiki.openstreetmap.org/wiki/Lifecycle_prefix
So why would you close yourself off to so much data? Why not give the community the chance to figure out which of the schemes would be preferred. By not showing an established second way of tagging, you are taking that decision out of the hands of the people.

Is it that big of a ressource problem to add the life cycle variants for tags?

**Comment by jeisenbe** [comment]
Please, if anyone would like to replace railway=disused with disused:railway=*, the place to do that would be at a wider, tagging-oriented discussion like the [tagging mailing list](https://lists.openstreetmap.org/listinfo/tagging). To deprecate a tag there is a process at https://wiki.openstreetmap.org/wiki/Proposal_process - if the proposal is approved, then first it would be best to request that editors like JOSM support the newly approved tag. This has more influence than what tags are rendered in this style. 

I've searched through the history of the relevant wiki pages, and from what I can tell the `disused:<key>` namespace was introduced to the [Key:disused=* wiki page in 2011](https://wiki.openstreetmap.org/w/index.php?title=Key%3Adisused%3A&type=revision&diff=642985&oldid=567060) without a proposal: . I don't know if it was widely discussed. 

This is not the place to change common ways of tagging features, but as I said above, we will happily reopen this issue if the wider Openstreetmap community decides to use `disused:railway=*` instead of `railway=disused`.

**Comment by dieterdreist** [comment]
Am Mi., 25. Sept. 2019 um 12:19 Uhr schrieb Christoph Hormann <
notifications@github.com>:

> I completely understand that there are people who would like to see this
> style take a more active role in steering mappers to map the 'right' way.
>

IMHO this is already what the style does. The project goal to act against
tag fragmentation and support only one tag for one thing, even if the
competing tag is used in significant numbers, is an example for this active
role that osm-carto plays in the tagging discussion.

**Comment by kocio-pl** [comment]
The most surprising fact for me is that @imagico is the most active person doing this (and trying to make it even more than it is today)... Christoph, I don't understand your position, it does not look consistent to me, what did you mean?

**Comment by kocio-pl** [comment]
> and support only one tag for one thing, even if the competing tag is used in significant numbers

There is no such rule in our goals, and if you mean "helps to prevent unfavorable fragmentation of tag use", it does not imply that.

**Comment by matkoniecz** [comment]
I would be happy to discuss issues raised here - both related to meaning of tags and more general strategy and properties of this map style or process of influencing tagging. But all of that is either offtopic here because it does not belong to an issue tracker or is more general than this specific issue.

How to handle competing tag schemes and at which point new scheme duplicating older one should be supported can be discussed but I think that it would be better to not hide discussion on this issue and base it on wider number of cases.

**Comment by kocio-pl** [comment]
Could you please open specific ticket then to separate the problems as much as you see it'd be useful?

**Comment by imagico** [comment]
As already said - if anyone wants to discuss changing overall goals and policy of this style this is welcome but there is a right place for that - which is not here, in a closed issue on a specific rendering question.
